client/src/main/java/io/pravega/client/tables/KeyVersion.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.tables;
+
+import java.io.Serializable;
+import lombok.Data;
+
+/**
+ * Version of a Key in a Table.
+ */
+@Data
+public class KeyVersion implements Serializable {
+    /**
+     * A special KeyVersion which indicates the Key must not exist when performing Conditional Updates.
+     */
+    public static final KeyVersion NOT_EXISTS = new KeyVersion(Long.MIN_VALUE, Long.MIN_VALUE);
```

Comment:
Yes that is one option. I preferred to keep the APIs to a minimum. The JavaDoc explains what these versions mean.

Reply:
Keeping the API surface in general to a minimum makes sense, but here one key point is clarity of the API and it is not really increasing the surface that much. If it is indifferent to you, could we do it in `TableSegment` the same that you did in `TableWriter` to be consistent?

Reply:
`TableSegment` is lower level than `TableWriter`. It has `put(single entry)` and `put(multiple entries)`. As such:
- If I want to add an API for `create`, I'd need to have `create(single entry)` and `create(multiple entries)`.
- Because of the "multiple entries" APIs, I need a `TableEntry` class, which needs to have a version. This same class would need to be reused for `create`, so it will make the API look weird if I need to create a key with a `TableEntry` that may or may not have a version. IMO, it will just increase the chances of writing bad code.
- Additionally, we want to be able to do batch updates, were we mix updating existing keys with creating (inserting) new keys. If we separate the APIs, we will not have the ability to perform that.

Yes, this API is different from `TableWriter`, however it provides more features and as such it is more complex and requires more careful programming against.